ruby_programming/advanced_ruby/blocks.md

```diff
@@ -475,24 +475,24 @@ After coming to grips with the information in this lesson you'll be a block, pro
 </div>
 
 ### Knowledge Check
- 
+
 This section contains questions for you to check your understanding of this lesson. If you're having trouble answering the questions below on your own, review the material above to find the answer.
- 
-- What is a block?
-- How is a block like a method?
-- How is a block different from a method?
-- What are the two ways to declare a block?
-- How do you return data from a block?
-- How can your methods collect the return data from a block?
-- What happens if you include a `return` statement in a block?
-- Why would you use a block instead of just creating a method?
-- What does `yield` do?
-- How do you pass arguments to a block from within a method?
-- How do you check whether a block was actually passed in?
-- What is a proc?
-- What is a lambda?
-- What's the difference between a proc and a block?
-- When would you use a proc instead of a block?
-- What's different between a lambda and a proc?
-- How do you convert a proc to a block?
-- How do you convert a block to a proc?
+
+- <a class="knowledge-check-link" href="https://www.rubyguides.com/2016/02/ruby-procs-and-lambdas/#Understanding_Ruby_Blocks">What is a block?</a>
+- <a class="knowledge-check-link" href="https://www.rubyguides.com/2016/02/ruby-procs-and-lambdas/#Understanding_Ruby_Blocks">How is a block like a method?</a>
+- <a class="knowledge-check-link" href="#block-basics">How is a block different from a method?</a>
+- <a class="knowledge-check-link" href="#block-basics">What are the two ways to declare a block?</a>
```

Comment:
I'm not understanding how these questions are answered in the linked section, can you explain that to me?

Reply:
The second question, line 484, is explicitly answered. The linked section states "A block can be declared as a single-line or multi-line block. Ruby convention is to use `{}` for single-line blocks and `do..end` for multi-line blocks."

I did not find a good answer to the first question, line 483, in the lesson or the linked materials. The linked information shows how to declare a block, which is one way the block and method differ.

Reply:
> The second question, line 484, is explicitly answered. The linked section states "A block can be declared as a single-line or multi-line block. Ruby convention is to use {} for single-line blocks and do..end for multi-line blocks."

Ah I see. The phrasing of the question was a bit confusing to me, I'll have to do some more thinking on if there's a different phrasing I would like to see.

Reply:
> I did not find a good answer to the first question, line 483, in the lesson or the linked materials. The linked information shows how to declare a block, which is one way the block and method differ.

True they are declared differently, but that doesn't feel right to me. I think that if there isn't a good answer in the lesson this question should maybe just be removed. The current direction we've gone with Knowledge Checks is a bit of a departure from how we used to do them, so sometimes the existing Knowledge Checks in a lesson won't be suited to stay.

Reply:
I am good with removing all of the questionable knowledge checks that you point out. Would it be better to make the changes and submit a new pull request or see if I can find a way to modify this one?

Reply:
Adding commits to an already open PR where changes have been requested is always the preferred method.

Reply:
I removed the confusing knowledge checks. I apologize for the extraneous commits. The git process didn't work quite like I expected and I ended up with two extra commits as I synced my branch with the main.

Reply:
No worries on the commits 👍